Factory/Headers/computer.h

```diff
@@ -0,0 +1,53 @@
+#ifndef COMPUTER_HPP
+#define COMPUTER_HPP
+
+class IComputer
+{
+public:
+	virtual void getComponentsInfo() = 0;
+}; // IComputer
+
+
+class PC : public IComputer
+{
+public:
+	PC();
+
+public:
+	void getComponentsInfo() override;
+
+private:
+	std::string m_CPU;
```

Comment:
You can add one level of abstraction (`class Computer : public IComputer`) and move common code (methods and fields) there to avoid code duplication